test/f3_test.go

```diff
@@ -0,0 +1,545 @@
+
+	// check that no manifest change has propagated
+	require.Equal(t, prev, env.nodes[0].f3.Manifest.Manifest())
+	env.requireEqualManifests(false)
+
+}
+
+func TestDynamicManifest_WithoutRebootstrap(t *testing.T) {
+	ctx := context.Background()
+	env := newTestEnvironment(t, 2, true)
+
+	env.Connect(ctx)
+	env.Run(ctx)
+	time.Sleep(1 * time.Second)
+
+	prev := env.nodes[0].f3.Manifest.Manifest()
+
+	env.waitForInstanceNumber(ctx, 3, 15*time.Second, false)
+	prevInstance := env.nodes[0].CurrentGpbftInstace(t, ctx)
+
+	env.manifest.BootstrapEpoch = 953
+	env.manifest.Sequence = 1
+	env.manifest.ReBootstrap = false
+	// Adding a power of 1 so we can reach consensus without having to run
+	// the new nodes
+	env.addPowerDeltaForParticipants(ctx, &env.manifest, []gpbft.ActorID{2, 3}, big.NewInt(1), false)
+	env.updateManifest()
+
+	env.waitForManifestChange(ctx, prev, 15*time.Second, []*testNode{env.nodes[0], env.nodes[1]})
+
+	// check that the runner continued without rebootstrap
+	require.True(t, env.nodes[0].CurrentGpbftInstace(t, ctx) >= prevInstance)
+	env.waitForInstanceNumber(ctx, prevInstance+10, 15*time.Second, false)
+	require.NotEqual(t, prev, env.nodes[0].f3.Manifest.Manifest())
+	env.requireEqualManifests(false)
+	// check that the power table is updated with the new entries
+	ts, err := env.ec.GetTipsetByEpoch(ctx, int64(env.nodes[0].CurrentGpbftInstace(t, ctx)))
+	require.NoError(t, err)
+	pt, err := env.nodes[0].f3.GetPowerTable(ctx, ts.Key())
+	require.NoError(t, err)
+	require.Equal(t, len(pt), 4)
+}
+
+var baseManifest manifest.Manifest = manifest.Manifest{
+	Sequence:        0,
+	BootstrapEpoch:  950,
+	ReBootstrap:     true,
+	InitialInstance: 0,
+	NetworkName:     gpbft.NetworkName("f3-test"),
+	GpbftConfig: &manifest.GpbftConfig{
+		// Added a higher delta and lower backoff exponent so tests run faster
+		Delta:                500 * time.Millisecond,
+		DeltaBackOffExponent: 1,
+		MaxLookaheadRounds:   5,
+	},
+	// EcConfig:        manifest.DefaultEcConfig,
+	EcConfig: &manifest.EcConfig{
+		ECFinality:       10,
+		CommiteeLookback: 5,
+		// increased delay and period to accelerate test times.
+		ECDelay:  500 * time.Millisecond,
+		ECPeriod: 500 * time.Millisecond,
+	},
+}
+
+type testNode struct {
+	h     host.Host
+	f3    *f3.F3
+	errCh <-chan error
+}
+
+func (n *testNode) CurrentGpbftInstace(t *testing.T, ctx context.Context) uint64 {
+	c, err := n.f3.GetLatestCert(ctx)
+	require.NoError(t, err)
+	if c == nil {
+		return 0
+	}
+	return c.GPBFTInstance
+}
+
+type testEnv struct {
+	t              *testing.T
+	manifest       manifest.Manifest
+	signingBackend *signing.FakeBackend
+	nodes          []*testNode
+	ec             *ec.FakeEC
+
+	manifestSender *passive.ManifestSender
+}
+
+// signals the update to the latest manifest in the environment.
+func (e *testEnv) updateManifest() {
+	e.manifestSender.UpdateManifest(&e.manifest)
+	// wait for the manifest to propagate
+	time.Sleep(2 * ManifestSenderTimeout)
```

Comment:
What indicates that the manifest has propagated? Wrap that logic in a `require.Eventually` instead of arbitrary wait time.